lib/codecs/UrlCodec.js

```diff
@@ -100,12 +103,45 @@ export class Base58DidUrlCodec {
     return decodedValue;
   }
 }
+const uuidRegex = /(urn:uuid):(.*)/g;
+export class UrnUuidCodec {
+  encode(value) {
+    // break the base58 DID URL into segments
+    const matches = Array.from(value.matchAll(uuidRegex));
+    const match = matches[0];
+
+    // encode the DID URL segments
+    const encodedUrl = (urlHeaders.has(match[1])) ? urlHeaders.get(match[1]) :
+      match[1];
```

Comment:
Can we simplify to this?

```suggestion
    const encodedUrl = urlHeaders.get(match[1]) || match[1];
```